### PR TITLE
Remove use of satori uuid package

### PR DESCRIPTION
**Pull request by Porges**
For #1291. Had to fix this with a go.mod rewrite as there are dependencies like the Go Azure SDK that depend on the satori package directly.

### CODE REVIEW

File: go.mod
Comment (Porges) [inline go.mod:7]: these other removals happened due to `go mod tidy`

**Review by matthchr — APPROVED** [review]
LGTM provided tests pass

**Comment by Porges** [comment]
@matthchr what's the test coverage like? are we safe?

**Comment by matthchr** [comment]
@Porges the test coverage isn't as good as it could/should be - but it's what we have.

I see this package is used by us in azuresqlmanageduser and keyvault. We definitely have tests for KeyVault and from what I can see at least `TestKeyvaultControllerWithAccessPolicies` should cover a path where this package is being used.

**Comment by Porges** [comment]
I might hold off on this for a bit while Azure SDK for Go updates their references. Then I shouldn’t need to do mod rewriting.

**Comment by Porges** [comment]
Tracking this PR in autorest: https://github.com/Azure/autorest.go/pull/520

**Pull request closed without merging**
